reaktive/src/commonMain/kotlin/com/badoo/reaktive/observable/Debounce.kt

```diff
@@ -1,66 +1,7 @@
 package com.badoo.reaktive.observable
 
-import com.badoo.reaktive.base.subscribeSafe
-import com.badoo.reaktive.disposable.CompositeDisposable
-import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.completable.completableTimer
 import com.badoo.reaktive.scheduler.Scheduler
-import com.badoo.reaktive.utils.atomic.AtomicReference
-import com.badoo.reaktive.utils.atomic.update
 
 fun <T> Observable<T>.debounce(timeoutMillis: Long, scheduler: Scheduler): Observable<T> =
-    observable { emitter ->
-        val disposables = CompositeDisposable()
-        emitter.setDisposable(disposables)
-        val executor = scheduler.newExecutor()
-        disposables += executor
-
-        subscribeSafe(
-            object : ObservableObserver<T> {
-                private val pendingValue = AtomicReference<DebouncePendingValue<T>?>(null, true)
-
-                override fun onSubscribe(disposable: Disposable) {
-                    disposables += disposable
-                }
-
-                override fun onNext(value: T) {
-                    val newPendingValue = DebouncePendingValue(value)
-                    pendingValue.value = newPendingValue
-
-                    executor.cancel()
-
-                    executor.submit(timeoutMillis) {
-                        pendingValue.update {
-                            if (it === newPendingValue) null else it
-                        }
-
-                        emitter.onNext(value)
-                    }
-                }
-
-                override fun onComplete() {
-                    executor.cancel()
-
-                    executor.submit {
-                        pendingValue
-                            .value
-                            ?.value
```

Comment:
This is actually bug, if `value` is null then `onNext` won't be called. Please add test for the case. For both variants.

Reply:
Should be fixed, please review